[Shawn Chew] iP

ShawnChew opened the pull request:
# DukePro

> "Your mind is for having ideas, not holding them." - David Allen ([source](https://dansilvestre.com/productivity-quotes))

DukePro frees your mind of having to remember things you need to do. It's,
- text-based
- easy to learn
- ~~FAST~~ *SUPER* FAST to use

All you need to do is,
1. download it from [here](https://nus-cs2103-ay2223s1.github.io/website/schedule/week4/project.html).
2. double-click it.
3. add your tasks.
4. let it manage your tasks for you :wink:

And it is **FREE**!

Features:
- [x] Managing tasks
- [  ] Managing deadlines (coming soon)
- [  ] Reminders (coming soon)

---

If you Java programmer, you can use it to practice Java too. Here's the `main` method:
```
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
        }
}
```

Commit:
Add Gradle support

ShawnChew committed:
Added functions for level 1

ShawnChew committed:
Edited functions for level 1

ShawnChew committed:
Functions to add and list, and reformatted the main code

ShawnChew committed:
Converted to Task objects

ShawnChew committed:
Added code to mark and unmark tasks

ShawnChew committed:
Added functions for todos, events and deadlines

ShawnChew committed:
Edited the printing style of mark and unmark

ShawnChew committed:
Added DukeException and code for handling error

ShawnChew committed:
Converted store to ArrayList

ShawnChew committed:
delete function added

ShawnChew committed:
updated input.txt and EXPECTED.txt

ShawnChew committed:
Fixed the error in the input.txt and EXPECTED.txt

ShawnChew committed:
Add load function to load data from hard disk when Duke starts up

ShawnChew committed:
Edit Deadline and Event Class to deal with dates

ShawnChew committed:
Merge branch 'branch-Level-7'

ShawnChew committed:
Merge branch 'branch-Level-8'

ShawnChew committed:
Fix no save issue

ShawnChew committed:
Update to use OOP

ShawnChew committed:
Put into packages

ShawnChew committed:
Merge branch 'add-gradle-support' of https://github.com/nus-cs2103-AY2223S1/ip

ReaganTan00 commented on src/main/java/duke/command/Command.java:
```diff
@@ -0,0 +1,168 @@
+
+    public abstract void execute(TaskList tasks, Ui ui, Storage storage) throws IOException;
+
+    public static Command of(String s) throws DukeException {
```
Not very clear what the method 'of' would do just based on naming choice.

ReaganTan00 commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,55 @@
+import duke.task.TaskList;
+import duke.ui.Ui;
+
+public class Duke {
```
Java documentation not done up yet.
No worries and good coding quality 👍

ReaganTan00 reviewed (commented):
Didn't see anything wrong with coding quality, good job 👍

ShawnChew committed:
Add JUnit tests

ShawnChew committed:
Package the App as a JAR file

ShawnChew committed:
Add find command

ShawnChew committed:
Tweak the code to comply with coding standard

ShawnChew committed:
Add Javadoc comments

ShawnChew committed:
Merge tag 'A-CodingStandard'

A-CodingStandard

ShawnChew committed:
Merge branch 'branch-A-CodingStandard'

ShawnChew committed:
Merge branch 'branch-Level-9'

ShawnChew committed:
Use CheckStyle

csheun commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,86 @@
+    /**
+     * Represents a Ui object.
+     */
+    private static Ui ui;
```
I like how you organise the code with OOP

csheun commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,86 @@
+     *
+     * @param args Ignored and unused command line arguments.
+     */
+    public static void main(String[] args) {
+        new Duke("data/duke.txt").run();
+    }
```
I like how you encapsulate all the details and logic within your class method.
This makes the main method much more concise. :)

csheun commented on src/main/java/duke/command/Command.java:
```diff
@@ -0,0 +1,279 @@
+    private static class ExitCommand extends Command {
+
+        @Override
+        public Boolean isExit() {
```
This naming sounds a little to me. Perhaps can change to canExit()?

csheun commented on src/main/java/duke/command/Command.java:
```diff
@@ -0,0 +1,279 @@
+package duke.command;
+
+import java.io.IOException;
+
+import java.util.List;
+import java.util.ArrayList;
+
+import duke.exception.DukeException;
+
+import duke.storage.Storage;
+
+import duke.task.Task;
+import duke.task.Todo;
+import duke.task.Deadline;
+import duke.task.Event;
+import duke.task.TaskList;
+
+import duke.ui.Ui;
```
Perhaps you can consider arranging your import statements in lexicographical order.

csheun reviewed (commented):
LGTM.
Keep up the good work, director!

ShawnChew committed:
Implement GUI with JavaFX

Shawn532 commented on src/main/java/duke/task/Event.java:
```diff
@@ -0,0 +1,52 @@
+    /**
+     * Contains the details of when the task will occur.
+     */
+    protected String at;
```
Not sure I like the variable name at. Mayble it can be clearer, like time?

Shawn532 commented on src/main/java/duke/task/Deadline.java:
```diff
@@ -0,0 +1,52 @@
+    /**
+     * Contains the details of when the task should be completed by.
+     */
+    protected String by;
```
I noticed the same issue in class Event too. Maybe a clearer name might be better.

Shawn532 commented on src/main/java/duke/task/Task.java:
```diff
@@ -0,0 +1,71 @@
+ * @version CS2103T AY 22/23 Sem 1
+ */
+public class Task {
+    /**
+     * The details of the task.
+     */
+    protected String description;
```
It is perfectly fine to write Javadoc of class members on three lines, but it can also be specified on a single line as follows.
```suggestion
    /** The details of the task. */
    protected String description;
```

Shawn532 commented on src/main/java/duke/command/Command.java:
```diff
@@ -0,0 +1,279 @@
+        }
+
+        @Override
+        public void execute(TaskList tasks, Ui ui, Storage storage) {
+            Task t = tasks.get(index);
+            switch (comm) {
+            case "mark":
+                t.mark();
+                ui.printMarked(t);
+                break;
+            case "unmark":
+                t.unMark();
+                ui.printUnMarked(t);
+                break;
+            }
+        }
+    }
```
Good job on no indentation for case clauses!

Shawn532 reviewed (commented):
Good job! The code follows coding standard very well. I can't find any naming, layout, statements errors. Javadocs are also written very clearly, and the format of Javadoc is perfect.

ShawnChew committed:
Fix visual bug for list command

ShawnChew committed:
Add assert statments

There are no assert statments in the code currently.

Assert statments should be added to help verify some
of the assumptions made for the code to work properly.

ShawnChew committed:
Set up CI

ShawnChew committed:
Merge pull request #1 from ShawnChew/branch-A-Assertions

Add assert statments

ShawnChew committed:
Merge branch 'master' of https://github.com/ShawnChew/ip

ShawnChew committed:
Improve Code Quality

Some of the codes are too long, and some of the codes do not
adhere to the coding standard. Edits made to make the code more
readable.

The code is refactored by,
*Renaming some of the variables
*Extract method

ShawnChew committed:
Edit CI file

ShawnChew committed:
Merge pull request #2 from ShawnChew/branch-A-CodeQuality

Branch a code quality

ShawnChew committed:
Remove echo method in Ui

ShawnChew committed:
Provide a way to attach priorities to item

ShawnChew committed:
Improve the GUI

ShawnChew committed:
Add Ui image

ShawnChew committed:
Add user guide

ShawnChew committed:
Delete duplicate file

ShawnChew committed:
Update readme file

ShawnChew committed:
Update readme file

ShawnChew committed:
Update readme file

ShawnChew committed:
Update readme file

ShawnChew committed:
Update readme file

ShawnChew committed:
Update readme file
